File: smashpen/src/main/java/com/smashpen/smashpen/domain/CharacterNotes.java
package com.smashpen.smashpen.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
public class CharacterNotes {

	@JsonIgnore @ManyToOne private User user;
	@Id @GeneratedValue private Long id;
	@ManyToOne private SmashCharacter smashCharacter;
	private String notes;

	CharacterNotes() {}

	public CharacterNotes(User user, SmashCharacter smashCharacter, String notes) {
		this.smashCharacter = smashCharacter;
		this.notes = notes;
		this.user = user;
	}

	public SmashCharacter getSmashCharacter() {
		return smashCharacter;
	}

	public void setSmashCharacter(SmashCharacter smashCharacter) {
		this.smashCharacter = smashCharacter;
	}

	public String getNotes() {
		return notes;
	}

	public void setNotes(String notes) {
		this.notes = notes;
	}

	public Long getId() {
		return id;
	}
}

File: smashpen/src/main/java/com/smashpen/smashpen/domain/CharacterNotes.kt
package com.smashpen.smashpen.domain

import com.fasterxml.jackson.annotation.JsonIgnore

import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class CharacterNotes {

    @JsonIgnore
    @ManyToOne
    var user: User

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = -1

    @ManyToOne
    var smashCharacter: SmashCharacter? = null

    var notes: String = ""

    constructor(user: User, smashCharacter: SmashCharacter, notes: String) {
        this.smashCharacter = smashCharacter
        this.notes = notes
        this.user = user
    }

}
